collect_app/src/main/java/org/odk/collect/android/formentry/ODKView.java

```diff
@@ -124,35 +127,27 @@
     PreferencesProvider preferencesProvider;
 
     @Inject
-    AudioRecorderViewModelFactory audioRecorderViewModelFactory;
+    ActivityAvailability activityAvailability;
+
+    private final WidgetFactory widgetFactory;
```

Comment:
It felt like it made sense to rearrange this relationship so the `ODKView` owns a `WidgetFactory` object. I think we probably want to think about caching objects that are created as dependencies for widgets (down the line ) and this helps with things like that.